limiter.go

```diff
@@ -45,15 +67,25 @@ func (l limiter) Acquire() bool {
 // AcquireWait waits for the resource to become available before returning.
 func (l limiter) AcquireWait() {
 	e := empty{}
-	l <- e
+	l.wait <- e
 }
 
 // Release returns the resource to the available pool.
 func (l limiter) Release() error {
 	select {
-	case <-l:
+	case <-l.wait:
 		return nil
 	default:
 		return fmt.Errorf("Release without an associated Acquire")
 	}
 }
+
+func (l limiter) pause() {
+	if l.maxPause <= 0 {
+		return
+	}
+	pauseTime := rand.Intn(int(l.maxPause / time.Millisecond))
```

Comment:
I wonder if we want to have some way to have a minimum pause time that isn't 0. but this is better than nothing.